## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 429517`

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/16691657667


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>wahay</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>wahay</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>wahay</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>wahay</summary>
<pre>
goroutine 25 [running]:
	goroutine running on other thread; stack unavailable
created by gopkg.in/check%2ev1.(*suiteRunner).forkCall in goroutine 7
	gopkg.in/check.v1@v1.0.0-20201130134442-10cb98267c6c/check.go:666 +0x235
FAIL	github.com/digitalautonomy/wahay/hosting	600.056s
=== RUN   Test
OK: 8 passed
--- PASS: Test (0.01s)
PASS
coverage: 100.0% of statements
ok  	github.com/digitalautonomy/wahay/test	0.028s	coverage: 100.0% of statements
=== RUN   Test
OK: 27 passed
--- PASS: Test (3.02s)
PASS
coverage: 56.7% of statements
ok  	github.com/digitalautonomy/wahay/tor	3.038s	coverage: 56.7% of statements
FAIL
make: *** [Makefile:80: test] Error 1</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>wahay</li>
  </ul>
</details>

--- a/pkgs/by-name/wa/wahay/package.nix
+++ b/pkgs/by-name/wa/wahay/package.nix
@@ -1,0 +1,98 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  gnumake,
+  pkg-config,
+  gtk3,
+  rubyPackages,
+  tor,
+  mumble,
+  installShellFiles,
+  wrapGAppsHook3,
+  writableTmpDirAsHomeHook,
+  nix-update-script,
+}:
+
+buildGoModule (finalAttrs: {
+  pname = "wahay";
+  version = "0.2";
+
+  src = fetchFromGitHub {
+    owner = "digitalautonomy";
+    repo = "wahay";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-59gVGDElIs+OQ/ELwOFjz6YqG18iHqoa2yYF2Ddi0/o=";
+  };
+
+  vendorHash = "sha256-w8lUnvy5dPMHWbzyyTq9Q/kE/4vSuOHffaY9CeasvQ0=";
+
+  buildInputs = [
+    gtk3
+  ];
+
+  nativeBuildInputs = [
+    gnumake
+    pkg-config
+    rubyPackages.sass
+    installShellFiles
+    wrapGAppsHook3
+  ];
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
+  ];
+
+  postPatch = ''
+    # Avoid Git invocation: Remove assignments starting with ‘$(shell git’
+    sed -E -i \
+      -e '/^\w+\s*[+:]?=\s*\$\(shell\s+git\>/d' \
+      Makefile
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make TAG_VERSION=v${finalAttrs.version} \
+         BUILD_TIMESTAMP="$(date -u -d "@$SOURCE_DATE_EPOCH" '+%Y-%m-%d %H:%M:%S')" \
+         build
+    runHook postBuild
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+    make test
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    installBin bin/wahay
+    installManPage packaging/ubuntu/ubuntu/usr/share/man/man1/wahay.1.gz
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : ${
+        lib.escapeShellArg (
+          lib.makeBinPath [
+            tor
+            mumble
+          ]
+        )
+      }
+    )
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    broken = stdenv.hostPlatform.isDarwin;
+    description = "Decentralized encrypted conference call application";
+    homepage = "https://wahay.org/";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ dvn0 ];
+    mainProgram = "wahay";
+  };
+})

